src/main/java/io/jenkins/plugins/checks/github/CheckRunGHEventSubscriber.java

```diff
@@ -26,77 +30,150 @@
-            LOGGER.log(Level.WARNING, ("Malformed repository URL: " + repoUrl).replaceAll("[\r\n]", ""));
-            return;
+        LOGGER.log(Level.INFO, "Received rerun request through GitHub checks API.");
+        try (ACLContext ignored = ACL.as(ACL.SYSTEM)) {
+            scheduleRerun(checkRun, payload);
         }
+    }
 
-        ACL.impersonate(ACL.SYSTEM, () -> {
-            final JSONArray pullRequests = json.getJSONObject("check_run").getJSONArray("pull_requests");
-            String branchName = "master";
-            if (pullRequests.size() != 0) {
-                branchName = "PR-" + pullRequests.getJSONObject(0).get("number");
-            }
+    private void scheduleRerun(final GHEventPayload.CheckRun checkRun, final String payload) {
+        final GHRepository repository = checkRun.getRepository();
+        final String branchName = getBranchName(JSONObject.fromObject(payload));
+
+        for (Job<?, ?> job : jenkinsFacade.getAllJobs()) {
+            Optional<GitHubSCMSource> source = gitHubSCMFacade.findGitHubSCMSource(job);
 
-            for (Job<?, ?> job : new JenkinsFacade().getAllJobs()) {
-                Optional<GitHubSCMSource> source = new GitHubSCMFacade().findGitHubSCMSource(job);
+            if (source.isPresent() && source.get().getRepoOwner().equals(repository.getOwnerName())
+                    && source.get().getRepository().equals(repository.getName())
+                    && job.getName().equals(branchName)) {
+                Cause cause = new GitHubChecksRerunActionCause(checkRun.getSender().getLogin());
+                ParameterizedJobMixIn.scheduleBuild2(job, 0, new CauseAction(cause));
 
-                if (source.isPresent() && source.get().getRepoOwner().equals(repository.getOwnerName())
-                        && source.get().getRepository().equals(repository.getName())
-                        && job.getName().equals(branchName)) {
-                    ParameterizedJobMixIn.scheduleBuild2(job, 0,
-                            new CauseAction(new GitHubChecksRerunActionCause()));
-                    break;
-                }
+                LOGGER.log(Level.INFO, String.format("Scheduled rerun (build #%d) for job %s, requested by %s",
+                        job.getNextBuildNumber(), jenkinsFacade.getFullNameOf(job),
+                        checkRun.getSender().getLogin()).replaceAll("[\r\n]", ""));
+                return;
             }
-        });
+        }
+
+        LOGGER.log(Level.WARNING, String.format("No proper job found for the rerun request from repository: %s and "
+                + "branch: %s", repository.getFullName(), branchName).replaceAll("\r\n", ""));
+    }
+
+    /**
+     * Get branch name from {@link JSONObject}.
+     *
+     * This method will be replaced by {@link CheckRunGHEventSubscriber#getBranchName(GHEventPayload.CheckRun, String)}
+     * after the release of github--api-plugin 1.116. The github-api has already released 1.116 and make
```

Comment:
so do we need this method? seems like it should be available?

Reply:
this method is available, but using the object from github-api is better since we don't have to parse the json again, and more safe.

the github-api has released 1.116, may just need to wait two or three days, the github-api-plugin will follow up

Reply:
kk